app/src/main/java/com/hamdan/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -19,182 +20,223 @@ class LoginViewModel @Inject constructor(
-                    val lastName: String = it.lastName
-                    val birthDay: String = it.birthDay
-                    val email: String = it.email
-                    val location: String = it.location
-                    createAccountState.value = CreateAccountState.Loading
-                    // TODO remove delay when login flow fully complete (FA-80 maybe)
-                    delay(1000)
-                    val code = try {
-                        createAccountRequestUseCase(
-                            firstName,
-                            lastName,
-                            birthDay,
-                            email,
-                            location
-                        )
-                    } catch (e: Exception) {
-                        // TODO implement enum rather than int in FA-80
-                        0
-                    }
-                    if (code in 200..299) {
-                        // TODO implement enum rather than int in FA-80
+            createAccountState.value.let {
+                val firstName = it.firstName
+                val lastName = it.lastName
+                val birthDay = it.birthDay
+                val email = it.email
+                val location = it.location
+                createAccountState.value = CreateAccountState(
+                    firstName = firstName,
+                    lastName = lastName,
+                    birthDay = birthDay,
+                    email = email,
+                    location = location,
+                    isLoading = true,
+                )
+                // TODO remove delay when login flow fully complete (FA-80 maybe)
+                delay(1000)
+                val result = createAccountRequestUseCase(
+                    firstName = firstName.text,
+                    lastName = lastName.text,
+                    birthDay = birthDay.text,
+                    email = email.text,
+                    location = location.text,
+                )
+                when (result) {
+                    CreateAccountResult.CREATE_SUCCESS -> {
+                        createAccountState.value = CreateAccountState()
                         // send to login page
-                        createAccountState.value = CreateAccountState.UserInputting()
                         onAccountCreateSuccess()
-                    } else if (code < 100) {
-                        // TODO implement enum rather than int in FA-80
-                        createAccountState.value = CreateAccountState.Error
-                    } else {
-                        // TODO implement enum rather than int in FA-80
-                        createAccountState.value = CreateAccountState.Error
-                        // maybe we want to prompt them with a dialog on what error was?
+                    }
+                    CreateAccountResult.CREATE_EXISTS -> {
+                        createAccountState.value = CreateAccountState(
```

Comment:
Instead of all these giant blocks of instantiating the same object, cant' you just copy from the existing one and only change the fields you need to change?

Reply:
but then I wouldn't be useing the cool plugin,
Updated to be all copys.